tests/test_breadth_first_search.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+
+from src.breadth_first_search import breadth_first_search
+
+
+class TestBreadthFirstSearch(unittest.TestCase):
+    def setUp(self):
+        self.fixture = {
+            "test": ["test1", "test2"],
+            "test1": ["test2", "test3", "found"],
+            "test2": ["test4"],
+            "test3": [],
+            "test4": [],
+            "found": []
+        }
```

Comment:
There is no pointer in search path to not sought node which is not first in array.

```
A: [B, ...] # pointer to B is first in array
B: [..., X] # X is not first in array, but it is value you are looking for
```

```
A: [B, ...]
B: [..., C] # pointer to C is not first in array — this case you forgot to test
C: [..., X]
```

Reply:
Sorry, man, I have noticed that graph for tests is not cyclic. `breadth_first_search` would pass all tests even though you didn't check that node haven't been in the path before